test/jasmine/tests/bar_test.js

```diff
@@ -423,6 +423,33 @@ describe('Bar.calc', function() {
         var cd = gd.calcdata;
         assertPointField(cd, 'mlw', [[2, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 10]]);
     });
+
+    it('should translate sizes e.g. 2000 to milliseconds not to year when base is present (vertical case)', function() {
+        var gd = mockBarPlot([{
+            type: 'bar',
+            base: [0],
+            y: [2000],
+            x: ['A']
+        }], {});
+
+        var cd = gd.calcdata;
+        assertPointField(cd, 'b', [[0]]);
+        assertPointField(cd, 's', [[2000]]);
+    });
```

Comment:
Don't these tests need to explicitly specify date axis type?

Reply:
Good catch. Fixed in https://github.com/plotly/plotly.js/pull/5061/commits/3e439e01428fb1a3d56aa45b7a51ed50a8dc6557.